### PR TITLE
Quality of life improvements for formula handling around BigInteger columns

### PR DESCRIPTION
**Pull request by jcferretti**
100% test coverage on new methods added.

image

### CODE REVIEW

**Comment by jcferretti** [comment]
Part of #1753

File: engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java [hunk 1]
```diff
@@ -20249,4 +20288,856 @@ public static boolean greaterEquals(BigDecimal a, float b) {
...
+            return null;
+        }
+        return BigInteger.valueOf(a).subtract(b);
+    }
+
+    public static BigDecimal minus(BigInteger a, double b) {
+        if (a == null || b == QueryConstants.NULL_DOUBLE) {
+            return null;
+        }
+        return new BigDecimal(a).subtract(BigDecimal.valueOf(b));
```
Comment (cpwright) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java:20436]: Won't this have a scale that is 0?

File: engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java [hunk 2]
```diff
@@ -20249,4 +20288,856 @@ public static boolean greaterEquals(BigDecimal a, float b) {
...
+
+    public static BigInteger multiply(byte a, BigInteger b) {
+        return multiply(b, a);
+    }
+
+    public static BigDecimal multiply(BigInteger a, double b) {
+        if (a == null || b == QueryConstants.NULL_DOUBLE) {
+            return null;
+        }
+        return new BigDecimal(a).multiply(BigDecimal.valueOf(b));
```
Comment (cpwright) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java:20515]: Won't this have a scale of zero?

File: engine/table/src/test/java/io/deephaven/engine/table/impl/lang/TestQueryLanguageFunctionUtils.java [hunk 3]
```diff
@@ -14048,4 +14049,686 @@ public static void test_bigdecimal_comps() {
...
+            sameBiOrBd(expected, result);
+        }
+    }
+
+    public static void test_biginteger_minus() {
+        final BigInteger bv1Left = BigInteger.valueOf(100);
+        final BigInteger bv1Right = BigInteger.valueOf(10);
+        final BigInteger bie = BigInteger.valueOf(90);
+        final BigDecimal bde = BigDecimal.valueOf(90);
+        final Object[] expectedAndresult = new Object[] {
```
Comment (cpwright) [inline engine/table/src/test/java/io/deephaven/engine/table/impl/lang/TestQueryLanguageFunctionUtils.java:14107]: You need tests with fractional doubles/floats and decimals.

File: engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java (hunk 2 above)
Comment (jcferretti) [reply engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageFunctionUtils.java:20515]: See previous answer to same.

File: engine/table/src/test/java/io/deephaven/engine/table/impl/lang/TestQueryLanguageFunctionUtils.java (hunk 3 above)
Comment (jcferretti) [reply engine/table/src/test/java/io/deephaven/engine/table/impl/lang/TestQueryLanguageFunctionUtils.java:14107]: Done

**Comment by jcferretti** [comment]
Unit test coverage remains at 100% for new code as of me writing this comment.

**Review by cpwright — APPROVED** [review]

**Pull request merged**